src/coreclr/jit/fginline.cpp

```diff
@@ -1075,7 +1140,8 @@ void Compiler::fgMorphCallInlineHelper(GenTreeCall* call, InlineResult* result,
 Compiler::fgWalkResult Compiler::fgFindNonInlineCandidate(GenTree** pTree, fgWalkData* data)
 {
     GenTree* tree = *pTree;
-    if (tree->gtOper == GT_CALL)
+    // We may get late devirtualization opportunities for virtual calls that are not inline candidates.
```

Comment:
I think `fgFindNonInlineCandidates` gets broken with these changes. We only want to run it the last time we visit a statement, and now we might run it multiple times.

Reply:
Resolved.